### PR TITLE
EasyBlock for VMD

### PR DESCRIPTION
**Pull request by thiell**
http://www.ks.uiuc.edu/Research/vmd/
Originally installed with EasyBuild 2.3

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86 up for giving this a review?

File: easybuild/easyblocks/v/vmd.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            raise EasyBuildError("Tcl is required to build VMD")
+
+        netcdfroot = get_software_root('netCDF')
+        if not netcdfroot:
+            raise EasyBuildError("netCDF is required to build VMD")
+
+        env.setvar('TCLLIB', '-F"%s"' % os.path.join(tclroot, 'lib'))
+        env.setvar('TCLINC', '-I"%s"' % os.path.join(tclroot, 'include'))
+
+        vmddir = os.path.join(self.builddir, "%s-%s" % (self.name.lower(), self.version))
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: `self.cfg['startdir']`?

File: easybuild/easyblocks/v/vmd.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        plugindir = os.path.join(vmddir, 'plugins')
+
+        os.mkdir(plugindir)
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: put this in a `try: except:` clause

File: easybuild/easyblocks/v/vmd.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            raise EasyBuildError("Tk is required to build VMD")
+        env.setvar('TK_INCLUDE_DIR', os.path.join(tkroot, 'include'))
+        env.setvar('TK_LIBRARY_DIR', os.path.join(tkroot, 'lib'))
+
+        pythonroot = get_software_root('Python')
+        if not pythonroot:
+            raise EasyBuildError("Python is required to build VMD")
+
+        # not so nice...
+        env.setvar('PYTHON_INCLUDE_DIR', os.path.join(pythonroot, 'include/python2.7'))
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: don't hard code the version

File: easybuild/easyblocks/v/vmd.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+    def build_step(self):
+
+        vmddir = os.path.join(self.builddir, "%s-%s" % (self.name.lower(), self.version))
+        vmdlibdir = os.path.join(vmddir, 'lib')
+        vmdsrcdir = os.path.join(vmddir, 'src')
+
+        # actc extracts to "actc-1.1"
+        actc_ver = self.cfg['actc_ver']
+        actcdir = os.path.join(self.builddir, "actc-%s" % actc_ver)
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: does it actually ship with multiple versions of `actc`?

File: easybuild/easyblocks/v/vmd.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+        try:
+            os.chdir(actcdir)
+        except OSError, err:
+            raise EasyBuildError("Could not chdir to {0}: {1}".format(actcdir, err))
+
+        actclibdir = os.path.join(vmdlibdir, 'actc')
+        shutil.rmtree(actclibdir, ignore_errors=True)
+        shutil.copytree(actcdir, actclibdir)
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: put this in the `try` clause of above

File: easybuild/easyblocks/v/vmd.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    def test_step(self):
+        pass
+
+    def sanity_check_step(self):
+        """Custom sanity check for VMD."""
+
+        custom_paths = {
+                        'files': ['bin/vmd'],
+                        'dirs': ['bin'],
+                       }
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: fix indenting here

File: easybuild/easyblocks/v/vmd.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+        super(EB_VMD, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_req_guess(self):
+
+        guesses = super(EB_VMD, self).make_module_req_guess()
+
+        guesses.update({
+            'PATH': ['bin'],
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: it doesn't find this by itself?

**Comment by boegel** [comment]
@thiell ping on looking into the remarks for this?

**Comment by akesandgren** [comment]
What's the status of this? We just got a request for VMD too.
@thiell If you don't have time i could probably take over moving this forward in a couple of days.

**Comment by boegel** [comment]
I have a need for this too now, so I should get to this soon.

I plan to submit a wrapper PR that includes this, and fixes the remarks on top.

**Comment by boegel** [comment]
enhanced/cleaned up version in #1148

**Pull request merged**